ogr/ogrsf_frmts/mssqlspatial/ogrmssqlspatialdatasource.cpp

```diff
@@ -690,6 +690,9 @@ int OGRMSSQLSpatialDataSource::Open( const char * pszNewName, bool bUpdate,
     char* pszGeometryFormat = nullptr;
     char* pszConnectionName = CPLStrdup(pszNewName + 6);
     char* pszDriver = nullptr;
+    char* pszUID = nullptr;
```

Comment:
those variables should also be free'd towards the end of the method

Reply:
See: [2db0d53](https://github.com/OSGeo/gdal/pull/6797/commits/2db0d533facb374a40a1adf34dc0613431bb3afa)